src/pages/NewsFeed.js

```diff
@@ -1,5 +1,45 @@
+import { useState, useEffect } from "react";
+import PostList from "../components/PostList";
+
 function NewsFeed() {
-  return <h1>News Feed</h1>;
+  const [isLoading, setIsLoading] = useState(true);
+  const [loadedPosts, setLoadedPosts] = useState([]);
+
+  useEffect(() => {
+    fetch(
+      "https://friendface-react-default-rtdb.europe-west1.firebasedatabase.app/posts.json"
+    )
+      .then((response) => {
+        return response.json();
+      })
```

Comment:
Do you need this point? Can't you combine it with the promise below and do data = response.json()?

Reply:
Was thinking as .json() returns a promise that I'd need to split it up and wait (using .then() ) for that to return the completed promise before working on it - what do you think?

Reply:
Hmm if json is a promise then this is fine. Maybe you could just call response.json() instead of creating a lambda to return it?

e.g.
.then(response.json()).then...

I can't remember if it should be a method reference (response.json) or it should be called (response.json()), try both and let me know what the behaviour is?